src/createNavigator.tsx

```diff
@@ -0,0 +1,14 @@
+import * as React from 'react';
+import { ParamListBase, ScreenProps, TypedNavigator } from './types';
+import Screen from './Screen';
+
+export default function createNavigator<ParamList extends ParamListBase>(
+  Navigator: React.ComponentType<any>
```

Comment:
I think we need to use generic for this for types to work properly

```
<ParamList extends ParamListBase, N extends React.ComponentType<any>>(Navigator: N): TypedNavigator<ParamList, N>
```

Though since this function is called by navigator now, we need to change it to not include the param list stuff.